hvplot/plotting/core.py

```diff
@@ -1882,9 +1882,12 @@ def _get_converter(self, x=None, y=None, kind=None, **kwds):
                     for v in params.values()
                     if isinstance(v, (str, list))
                 ]
+
                 columns = (
                     set(self._data.columns) & set(itertools.chain(*possible_columns))
                 ) or {self._data.columns[0]}
```

Comment:
```suggestion
                ) or {next(c for c in self._data.select(pl.col(pl.NUMERIC_DTYPES)).columns if c != x or c not in x)}
```

I think the logic of selecting the first column is what is wrong. Can you try this?

Reply:
@Hoxbro that would still drop numeric cols, no? It'll just get the first col that isn't `x`

Reply:
Yeah, you are right. I just took a quick look at your example code in the issue. To get parity with pandas, we should get all the numeric columns like what you are doing in this PR. (This does not give the same result with polars as it does with pandas `DataFrame({"a": [1, 2], "b": [3, 4], "c": [5, 6]})` with my suggested change.)

I think we should only add all numeric columns if `y is None` because if `x is None` we will get a range from 0 to the size of the column.

Reply:
Yes you are right, I had misread the docs for the x param. Pushed another commit